clients/drcachesim/tests/burst_aarch64_sys.cpp

```diff
@@ -0,0 +1,265 @@
+        std::cerr << "failed to set env var!\n";
+
+    std::cerr << "pre-DR init\n";
+    dr_app_setup();
+    assert(!dr_app_running_under_dynamorio());
+    std::cerr << "pre-DR start\n";
+    dr_app_start();
+    assert(dr_app_running_under_dynamorio());
+    do_some_work();
+    dr_app_stop_and_cleanup();
+    std::cerr << "all done\n";
+    assert(!dr_app_running_under_dynamorio());
+    return post_process();
+}
+
+bool
+is_dc_zva_instr(void *dr_context, memref_t memref)
+{
+    if (!type_is_instr(memref.instr.type))
+        return false;
+    app_pc pc = (app_pc)memref.instr.addr;
+    instr_t instr;
+    instr_init(dr_context, &instr);
+    auto *ret = decode(dr_context, pc, &instr);
+    assert(ret != NULL && instr_valid(&instr));
+    bool is_dc_zva = instru_t::is_aarch64_dc_zva_instr(&instr);
+    instr_free(dr_context, &instr);
+    return is_dc_zva;
+}
+
+int
+main(int argc, const char *argv[])
+{
+    // App setup.
+    signal(SIGILL, sigill_handler);
+
+    // Gather app trace.
+    std::string trace_dir = gather_trace();
+
+    void *dr_context = dr_standalone_init();
+    analyzer_t analyzer(trace_dir);
+    if (!analyzer) {
+        std::cerr << "Failed to initialize iterator " << analyzer.get_error_string()
+                  << "\n";
+    }
+    bool found_cache_line_size_marker = false;
+    int dc_zva_instr_count = 0;
+    int dc_zva_memref_count = 0;
+    addr_t last_dc_zva_pc = 0;
+    for (reader_t &iter = analyzer.begin(); iter != analyzer.end(); ++iter) {
+        memref_t memref = *iter;
+        if (memref.marker.type == TRACE_TYPE_MARKER &&
+            memref.marker.marker_type == TRACE_MARKER_TYPE_CACHE_LINE_SIZE) {
+            found_cache_line_size_marker = true;
+            assert(memref.marker.marker_value == proc_get_cache_line_size());
+        }
+        if (is_dc_zva_instr(dr_context, memref)) {
+            dc_zva_instr_count++;
+            last_dc_zva_pc = memref.instr.addr;
+            continue;
+        }
+        // Look for _memref_data_t entries.
+        if ((memref.data.type == TRACE_TYPE_READ ||
+             memref.data.type == TRACE_TYPE_WRITE || type_is_prefetch(memref.data.type))
+            // Look for memrefs for last seen dc zva pc.
+            && (last_dc_zva_pc != 0 && memref.data.pc == last_dc_zva_pc)) {
+            assert(memref.data.type == TRACE_TYPE_WRITE);
+            dc_zva_memref_count++;
+            assert(ALIGNED(memref.data.addr, proc_get_cache_line_size()));
+            assert(memref.data.size == proc_get_cache_line_size());
+        }
+    }
+    std::cerr << "DC ZVA count: " << dc_zva_instr_count << "\n";
+    std::cerr << "DC ZVA memref count: " << dc_zva_memref_count << "\n";
+    std::cerr << "All DC ZVA memrefs are cache-line shaped stores!\n";
+    assert(dc_zva_memref_count != 0);
+    assert(dc_zva_instr_count == dc_zva_memref_count);
+    assert(found_cache_line_size_marker);
+    dr_standalone_exit();
+
+    return 0;
+}
```

Comment:
This comment is for clients/drcachesim/tests/drmemtrace.threadsig.aarch64.raw/libdrmemtrace.so 

This file should not be changed.

I added a README here on how to regenerate these files:

https://github.com/DynamoRIO/dynamorio/pull/4447

Reply:
Please review the README and update the files here: key things are `-Os -static`, `4 2000`, synthetic libs.

Reply:
Thanks for the steps.

I've stripped the libdynamorio.so binary. It reduced the size a bit, ~2%. 

The args to threadsig that I used were already the same as you mentioned (I guessed the second arg based on the existing counts in test template). So, no change needed in the traces.